Projects/LinkedListIterator/LinkedListIterator/LinkedListIterator/DLList.h

```diff
@@ -0,0 +1,398 @@
+	}
+
+	//-------------------------------------------------------------------------------------------
+	//											SIZE
+	//-------------------------------------------------------------------------------------------
+	bool IsEmpty() const
+	{
+		return Size == 0;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	int GetSize() const
+	{
+		return Size;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	//										MODIFIERS
+	//-------------------------------------------------------------------------------------------
+	void AddHead(const T& NewData)
+	{
+		FNode* Node = new FNode;
+		Node->Data = NewData;
+		Node->Next = Head;
+
+		if (!Head)
+		{
+			Tail = Node;
+		}
+		else
+		{
+			Head->Previous = Node;
+		}
+		Head = Node;
+		++Size;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void AddTail(const T& NewData)
+	{
+		if (Tail)
+		{
+			FNode* Node = new FNode;
+			Node->Data = NewData;
+			Node->Previous = Tail;
+			Tail->Next = Node;
+			Tail = Node;
+
+			++Size;
+		}
+		else
+		{
+			AddHead(NewData);
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void Insert(const T& NewData, const int Index)
+	{
+		if (Index < 0 || Index > Size) //Bounds Checking
+		{
+			return;
+		}
+		else if (Index == 0) //Head
+		{
+			AddHead(NewData);
+		}
+		else if (Index == Size) //Tail
+		{
+			AddTail(NewData);
+		}
+		else if (Index < Size)
+		{
+			FNode* Current = GetNode(Index);
+			FNode* NewNode = new FNode;
+
+			NewNode->Data = NewData;
+			NewNode->Next = Current;
+			NewNode->Previous = Current->Previous;
+
+			Current->Previous->Next = NewNode;
+			Current->Previous = NewNode;
+
+			Size++;
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void RemoveHead()
+	{
+		FNode* NewHead = Head->Next;
+		delete Head;
+		Head = NewHead;
+
+		if (Size == 1)
+		{
+			Tail = nullptr;
+		}
+		else
+		{
+			Head->Previous = nullptr;
+		}
+		--Size;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void Remove(const int Index)
+	{
+		if (Index < 0 || Index >= Size) //Bounds Checking
+		{
+			return;
+		}
+		else if (Index == 0) //Head
+		{
+			RemoveHead();
+		}
+		else
+		{
+			FNode* Current = GetNode(Index);
+			Current->Previous->Next = Current->Next;
+
+			if (Index == Size - 1)
+			{
+				Tail = Current->Previous;
+			}
+			else
+			{
+				Current->Next->Previous = Current->Previous;
+			}
+			delete Current;
+			--Size;
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void Clear()
+	{
+		while (Head)
+		{
+			RemoveHead();
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	//										ITERATOR
```

Comment:
maybe you should put this closer to the class FIterator, just to sort your class, furthermore, you could make your class more readable and you can have all elements related to the iterator in the same code chunk